### PR TITLE
docs: improve analyze() docstring with full runnable example

### LINKED ISSUE #22: Improve docstring for analyze() with a full code example
# [DOCS] Improve `analyze()` docstring with a full runnable example

**Difficulty:** ⭐ Beginner  
**Estimated Time:** 15–20 minutes  

Labels: `good first issue`, `docs`, `beginner-friendly`

---

## 🔍 Problem

The current `analyze()` docstring includes only a brief example.

For new users, it’s not immediately clear how to:
- create a dataset  
- train a model  
- run `analyze()` end-to-end  

👉 We should provide a **complete, runnable example** directly in the docstring.

---

## 🎯 Task

Expand the `Examples` section of the `analyze()` function docstring.

Add a **clear, 10-line runnable example** that demonstrates:
- dataset creation using `make_classification`
- training a simple model
- running `analyze()`
- displaying results

---

## 🛠️ Where to make changes

- File: `trustlens/api.py`
- Function: `analyze()`

---

## 🧭 Getting Started

1. Open `trustlens/api.py`
2. Locate the `analyze()` function
3. Find the existing docstring
4. Navigate to the `Examples` section
5. Replace or expand it with a full working example

---

## 📌 Example (target style)

```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from trustlens import analyze

# Create dataset
X, y = make_classification(n_samples=500, n_features=10, random_state=42)

# Train/test split
X_train, X_test, y_train, y_test = train_test_split(X, y)

# Train model
model = RandomForestClassifier()
model.fit(X_train, y_train)

# Predict probabilities
y_prob = model.predict_proba(X_test)

# Run analysis
report = analyze(model, X_test, y_test, y_prob)

# Show results
report.show()
````

---

## ✅ Expected Outcome

* The docstring contains a **complete, copy-paste runnable example**
* Users can quickly understand how to use `analyze()`
* Improves developer experience and onboarding

---

## 🧪 How to Verify

* Open the updated docstring in your editor
* Copy the example code and run it in a Python environment
* Confirm it runs without errors and produces output

---

## 🤝 First-time contributor?

If this is your first open-source contribution, comment:

> "I'd like to work on this!"

I’ll guide you step-by-step 🚀

### PR DESCRIPTION
**Pull request by q404365631**
## Summary

Closes #22

Added a complete, copy-paste runnable example to the `analyze()` docstring that demonstrates:

- Dataset creation using `make_classification`
- Train/test split
- Training a `RandomForestClassifier`
- Predicting probabilities
- Running `analyze()`
- Displaying results with `report.show()`

## Changes

- `trustlens/api.py`: Added `Examples` section to `analyze()` docstring with a full end-to-end runnable example

## Testing

- The example code follows the exact pattern suggested in the issue
- Uses standard sklearn imports that are already project dependencies
- Follows NumPy docstring conventions for the `Examples` section

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive usage examples to the `analyze` function documentation, including an end-to-end workflow demonstrating dataset creation, model training, probability prediction, and function invocation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Khanz9664/TrustLens/pull/112?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by Khanz9664 — APPROVED** [review]
@q404365631 Thanks for putting this together — this is a nice improvement to the developer experience.

I like that the example is fully runnable end-to-end instead of being pseudo-code. Showing dataset creation → train/test split → model training → `predict_proba()` → `analyze()` → `report.show()` makes the onboarding path much clearer for new users.

The scope is also nice and focused (docstring only, no unrelated changes), and everything looks clean with CI passing.

One small non-blocking thought for future polish: depending on how much automatic inference we want to showcase in TrustLens, we may eventually simplify the example if `analyze()` already handles probability generation internally for supported sklearn models. But the current example is still correct and very helpful as-is.

Nice contribution — thanks for improving the docs 🚀
